### ISSUE TITLE
Warnings during compilation of CMSSW environment

### ISSUE THREAD

**Issue by athompson-git** [cms-mual/Alignment#27]
When I run the following commands (1) (taken directly from the 10_0_0_pre3 branch of cms-mual/Alignment), I get the warnings below during compilation (2).

(1)
```
export RELEASE=CMSSW_10_0_0_pre3
SCRAM_ARCH=slc6_amd64_gcc630; export SCRAM_ARCH;
cmsrel $RELEASE
cd $RELEASE/src/
cmsenv

git clone https://github.com/cms-mual/Alignment.git -b $RELEASE
git clone https://github.com/cms-mual/TrackingTools.git -b $RELEASE
git clone https://github.com/cms-mual/MuAlSupplementaryFiles.git -b CMSSW_10_0_X
cp MuAlSupplementaryFiles/* .

ln -s Alignment/MuonAlignmentAlgorithms/scripts/createJobs.py
ln -s Alignment/MuonAlignmentAlgorithms/python/gather_cfg.py
ln -s Alignment/MuonAlignmentAlgorithms/python/align_cfg.py
ln -s Alignment/MuonAlignmentAlgorithms/scripts/submitBatchJobs.py
ln -s Alignment/MuonAlignmentAlgorithms/scripts/runBatchJobMonitor.py

scram b -j8
```

(2) The only warnings that appeared are the ones below, so I have truncated the full compilation stdout for the sake of readability

> Compiling  
```
/afs/cern.ch/work/a/adthomps/public/CMSSW_10_0_0_pre3/src/Alignment/MuonAlignmentAlgorithms/src/MuonResidualsPositionFitter.cc
/afs/cern.ch/work/a/adthomps/public/CMSSW_10_0_0_pre3/src/Alignment/MuonAlignmentAlgorithms/src/MuonResidualsFromTrack.cc: In constructor 'MuonResidualsFromTrack::MuonResidualsFromTrack(const edm::EventSetup&, edm::ESHandle<MagneticField>, edm::ESHandle<GlobalTrackingGeometry>, edm::ESHandle<DetIdAssociator>, edm::ESHandle<Propagator>, const Trajectory*, const reco::Track*, AlignableNavigator*, double, bool, bool, DTLayerData*, TTree*, CSCLayerData*, TTree*)':
/afs/cern.ch/work/a/adthomps/public/CMSSW_10_0_0_pre3/src/Alignment/MuonAlignmentAlgorithms/src/MuonResidualsFromTrack.cc:121:37: warning: unused variable 'tsosF' [-Wunused-variable]
     const TrajectoryStateOnSurface& tsosF = trajMeasurement.forwardPredictedState();
                                     ^~~~~
/afs/cern.ch/work/a/adthomps/public/CMSSW_10_0_0_pre3/src/Alignment/MuonAlignmentAlgorithms/src/MuonResidualsFromTrack.cc:122:37: warning: unused variable 'tsosB' [-Wunused-variable]
     const TrajectoryStateOnSurface& tsosB = trajMeasurement.backwardPredictedState();
                                     ^~~~~
/afs/cern.ch/work/a/adthomps/public/CMSSW_10_0_0_pre3/src/Alignment/MuonAlignmentAlgorithms/src/MuonResidualsFromTrack.cc:123:37: warning: unused variable 'tsosU' [-Wunused-variable]
     const TrajectoryStateOnSurface& tsosU = trajMeasurement.updatedState();
                                     ^~~~~
```

**Comment by demarley** [comment]
edited formatting

**Comment by athompson-git** [comment]
These same warnings appear for me when calling scram b on 9_4_0, so I don't think this is release specific; could it have to do with my lxplus?

**Comment by demarley** [comment]
I don't think so.  I don't remember getting these warnings so I can't say more until I have a chance to investigate further.

**Comment by lpernie** [comment]
In your error you are in CMSSW_10_0_0_pre2 but the error come from CMSSW_10_0_0_pre3
In which release are you?

**Comment by demarley** [comment]
Maybe a copy-paste error?  
@athompson-tamu you said there was another issue with `10_0_0_pre2` right?

**Comment by athompson-git** [comment]
Oops! I am working in pre3. It's just that the README for pre3 was not updated to say "export RELEASE=CMSSW_10_0_0_pre3" which I made sure to fix when I created my cms release, but I forgot to change it when I wrote this issue.

@demarley this was the only issue

Edit: fixed initial comment.

**Comment by demarley** [comment]
The warnings are from these [lines](https://github.com/cms-mual/Alignment/blob/CMSSW_10_0_0_pre3/MuonAlignmentAlgorithms/src/MuonResidualsFromTrack.cc#L121-L123).  
The variables aren't actually used anymore -- they were only loaded for a `DEBUG` [statement](https://github.com/cms-mual/Alignment/blob/CMSSW_10_0_0_pre3/MuonAlignmentAlgorithms/src/MuonResidualsFromTrack.cc#L235-L237), but that has since been commented out.  I've removed them.

**Comment by demarley** [comment]
Addressed by #29
